kategory/src/main/kotlin/kategory/typeclasses/MonadReader.kt

```diff
@@ -11,5 +11,11 @@ interface MonadReader<F, D> : Monad<F> {
     fun <A> reader(f: (D) -> A): HK<F, A> = map(ask(), f)
 }
 
+inline fun <reified F, A, reified D> HK<F, A>.local(FT: MonadReader<F, D> = monadReader(), noinline f: (D) -> D): HK<F, A> =
```

Comment:
Assuming Option has a monadReader, could you call `1.some().local { it }`? I thought defaults only worked if they were the last parameters.